### DISCUSSION TITLE
Not compiling in vscode in windows

### DISCUSSION THREAD

**Discussion by vascoosx** [jbangdev/jbang#1076] [General]
Hi,

I am running jbang v0.81.2 and recently had an error while trying to run in windows 10.
The error says "Main method not found in the file, please define the main method as: public static void main(String[] args)".

I have attempted to rebuild the workspace with "alt-shift B" (option Full) but the problem persisted. When I deleted the .class file in the "bin" directory it was not created again after attempting to rebuild with "alt-shift B". Therefore I concluded that it is not recompiling.

I also attempted to recreate everything with `jbang edit --fresh --no-open ...` but it gives me the same directory. So I'm not sure this is really fresh.

**Comment by quintesse** [comment]
First, if you really want to be sure you get a new/fresh directory you can actually clear the project cache by running `jbang cache clear --project`, but that will remove all projects so make sure you don't have any open in any IDEs. But you can also manually delete the project folder yourself if you want.

Second, if your IDE is compiling or not doesn't affect JBang, it does its own compilation, `jbang yourfile.java` should either just work or give you a compile error. If it doesn't seem to pick up the changes you've made to the file you _can_ try running with the `--fresh` option, but that really shouldn't be necessary.

And finally, given the error are you sure you have a proper `main()` method in your code?

**Comment by vascoosx** [comment]
Thank you for your advice. The code was running in the beginning but it stopped running after I've changed some things but I didn't delete the `main()` method.

I haven't deleted the directory in order to be able to investigate it.

I was able to create the jar file. I will see if the jar file runs as expected to check that the code doesn't have any issue.

**Reply by maxandersen** [reply]
Did it work for you ?

**Reply by vascoosx** [reply]
Sorry, can’t test it right now. I can test it on Monday.

**Reply by vascoosx** [reply]
Executing the jar file also produced the same result, that the main function is not present. However I do see the main function.

```java
    public static void main(String... args) throws Exception {
        int exitCode = new CommandLine(new CdxmlFixer()).execute(args);
        System.exit(exitCode);
    }
```

**Reply by maxandersen** [reply]
Can you share the full source?

**Reply by vascoosx** [reply]
Seems like the problem was that I wasn't running it with my admin account. Currently I cannot create a symbolic link with my normal account so I have to use my admin account to do the `jbang edit`. I think I used to create the IDE project with my admin account and then edit it with my normal account but apparently that's not possible, at least with this version.

**Reply by maxandersen** [reply]
Ah yeah - did jbang edit not report something ? 

It should stop if failure to link or at least warn - it still should work if it completed.

**Reply by vascoosx** [reply]
Yes, if I try to `jbang edit` with my normal account I get a warning that it cannot create a symbolic link and it fails to launch. So I was trying to create the folder with the `--no-open` option using my admin account, then, with my normal account, I would copy the path and run `code /temporary/project/path`. 

I was somehow very sure that even when creating with the admin account, I could use the resulting project with my normal account. 
However since this is not the case, I'll use only my admin account from now on. Alternatively I'll use wsl with my normal account because sometimes there are some issues when developing with my admin account. Plus, I can still run things in wsl from the host so maybe this is the best way for me or someone who has their admin account separate from their normal account like myself.

**Reply by quintesse** [reply]
@vascoosx It's a bit of a work-around, but to make sure your non-admin user has proper rights to create symbolic links you can install [Link Shell Extension](https://schinagl.priv.at/nt/hardlinkshellext/linkshellextension.html). It's what I did on a system of mine because Microsoft's documentation on how to do it yourself is pretty confusing. After installing that program your user should have all the proper rights and JBang should work okay, (You might need to deleted the project folder created by your admin user, though, just to make sure there's no conflict)
